### PR TITLE
add metadata feilds including labels, annotations, uid and image

### PR DESCRIPTION
**Pull request by wanlonghenry**
add metadata feilds including labels, annotations, uid and image
default include all the feilds

### CODE REVIEW

File: build/common/installer/scripts/tomlparser.rb [hunk 1]
```diff
@@ -211,6 +212,7 @@ def populateSettingValuesFromConfigMap(parsedConfig)
     begin
       if !parsedConfig[:log_collection_settings][:metadata_collection].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:enabled].nil?
         @logEnableKubernetesMetadata = parsedConfig[:log_collection_settings][:metadata_collection][:enabled]
+        @logKubernetesMetadataiIncludeFields = parsedConfig[:log_collection_settings][:metadata_collection][:include_fields]
```
Comment (pfrcks) [inline build/common/installer/scripts/tomlparser.rb]: pls add a nil check. if a customer does not specify the list then we should not collect any metadata

File: build/common/installer/scripts/tomlparser.rb [hunk 2]
```diff
@@ -25,6 +25,7 @@
 @adxDatabaseName = "containerinsights" # default for all configurations
 @logEnableMultiline = "false"
 @logEnableKubernetesMetadata = false
+@logKubernetesMetadataiIncludeFields = "podLabels,podAnnotations,poduid,image"
```
Comment (pfrcks) [inline build/common/installer/scripts/tomlparser.rb]: podUID

Comment (wanlonghenry) [reply build/common/installer/scripts/tomlparser.rb]: change all to podUid to sync with othe places: 
image

File: build/common/installer/scripts/tomlparser.rb (hunk 1 above)
Comment (wanlonghenry) [reply build/common/installer/scripts/tomlparser.rb]: added

File: build/common/installer/scripts/tomlparser.rb [hunk 3]
```diff
@@ -209,8 +210,9 @@ def populateSettingValuesFromConfigMap(parsedConfig)
 
     #Get Kubernetes Metadata setting
     begin
-      if !parsedConfig[:log_collection_settings][:metadata_collection].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:enabled].nil?
+      if !parsedConfig[:log_collection_settings][:metadata_collection].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:enabled].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:include_fields].nil?
         @logEnableKubernetesMetadata = parsedConfig[:log_collection_settings][:metadata_collection][:enabled]
+        @logKubernetesMetadataiIncludeFields = parsedConfig[:log_collection_settings][:metadata_collection][:include_fields]
```
Comment (pfrcks) [inline build/common/installer/scripts/tomlparser.rb]: looks good however we might be missing the default scenario.

1) customer enables metadata collection and does not specify field -> default scenario where we collect all field
2) customer enables metadata collection and does specify field -> we use the include_fields property.

as of now only option 2 is tackled

Comment (wanlonghenry) [reply build/common/installer/scripts/tomlparser.rb]: I think you are talking about the scenario where enabled is true but include_fields is commented out. We can add a scenario where enabled is true and parsedConfig[:log_collection_settings][:metadata_collection][:include_fields] is nil.

File: build/common/installer/scripts/tomlparser.rb [hunk 4]
```diff
@@ -211,6 +212,8 @@ def populateSettingValuesFromConfigMap(parsedConfig)
     begin
       if !parsedConfig[:log_collection_settings][:metadata_collection].nil? && !parsedConfig[:log_collection_settings][:metadata_collection][:enabled].nil?
         @logEnableKubernetesMetadata = parsedConfig[:log_collection_settings][:metadata_collection][:enabled]
+        if !parsedConfig[:log_collection_settings][:metadata_collection][:include_fields].nil?
+          @logKubernetesMetadataiIncludeFields = parsedConfig[:log_collection_settings][:metadata_collection][:include_fields]
```
Comment (pfrcks) [inline build/common/installer/scripts/tomlparser.rb:216]: missing end for the if block. also add a puts statement inside the block "config::Using config map setting for kubernetes metadata include fields"

File: build/common/installer/scripts/tomlparser.rb [hunk 5]
```diff
@@ -211,6 +212,8 @@ def populateSettingValuesFromConfigMap(parsedConfig)
...
         puts "config::Using config map setting for kubernetes metadata"
```
Comment (pfrcks) [inline build/common/installer/scripts/tomlparser.rb:217]: shift this line to above line 214

**Pull request merged**